src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -0,0 +1,235 @@
+  */
+  void runAll(){
+    int Total_number_of_tests = 0;
+    int Successful_Tests = 0;
+    int Failed_Tests = 0;
+    ArrayList<String> Failed_Tests_List = new ArrayList<String>();
+    
+    /*String InfrastructurePath = "../../../infrastructure/tests";
+    File Infrastucture = new File(InfrastructurePath);
+    String[] Tests1 = Infrastucture.list();
+
+    String ContentPath = "../../../content/tests";
+    File Content = new File(ContentPath);
+    String[] Tests2 = Content.list();
+
+    String NetworkPath = "../../../network/tests";
+    File Network = new File(NetworkPath);
+    String[] Tests3 = Network.list();
+
+    String UIPath = "../../../UI/tests";
+    File UI = new File(UIPath);
+    String[] Tests4 = UI.list();*/
+
+    String[][] Tests = getAllTests();
+
+    
+
+    for(int i =0; i < 4; i++){
+      for(int j = 0; j<Tests[i].length;j++){
+        boolean result = runByName(Tests[i][j]);
+        if(result){
+          Successful_Tests++;
+        }
+        else{
+          Failed_Tests_List.add(Tests[i][j]);
+          Failed_Tests++;
+
+        }
+        Total_number_of_tests++;
+      }
+    }
+
+
+
+    /*//get all the test cases in infrastucture module
+    File testFolderInfrastucture = new File("../../../infrastucture/tests");
+    File[] listOfTests1 = testFolderInfrastucture.listFiles();
+    System.out.println("Running all the test cases of infrastucture module...");
+    for(File file : listOfTests1){
+      if(file.isFile()){
+        runByName(file.getName());
+      }
+    }
+    
+    //get all the test cases in networking module
+    File testFolderNetworking = new File("../../../networking/tests");
+    File[] listOfTests2 = testFolderNetworking.listFiles();
+    System.out.println("Running all the test cases of networking module...");
+    for(File file : listOfTests2){
+      if (file.isFile()){
+        runByName(file.getName());
+      }
+    }
+
+    //get all the test cases in processing module
+    File testFolderProcessing = new File("../../../processing/tests");
+    File[] listOfTests3 = testFolderProcessing.listFiles();
+    System.out.println("Running all the test cases of processing module...");
+    for(File file : listOfTests3){
+      if(file.isFile()){
+        runByName(file.getName());
+      }
+    }
+    
+    //get all the test cases in UI module
+    File testFolderUI = new File("../../../UI/tests");
+    File[] listOfTests4 = testFolderUI.listFiles();
+    System.out.println("Running all the test cases of UI module...");
+    for(File file : listOfTests4){
+      if(file.isFile()) {
+        runByName(file.getName());
+      }
+    }
+    */
+}
+
+public String[][] getAllTests(){
+  String InfrastructurePath = "../../../infrastructure/tests";
+    File Infrastucture = new File(InfrastructurePath);
+    String[] Tests1 = Infrastucture.list();
+
+    String ContentPath = "../../../content/tests";
+    File Content = new File(ContentPath);
+    String[] Tests2 = Content.list();
```

Comment:
content lies under infrastructure. Both validation(Which is where test harness is) and content are inside infra